### ISSUE TITLE
SUGGESTION: Address data type

### ISSUE THREAD

**Issue by bobvanluijt** [semi-technologies/weaviate#1081]
Suggestion to add an address data-type which automatically parses the address.

## Expected behavior

1. When a user adds a thing or action schema item, they can set a data-type.
2. If a user creates a thing or action, they will do this as a string `{ "address": "Spuistraat 210, 1012 VT Amsterdam" }`
3. Weaviate parses the result and returns an address object:

```json
{
    "address": {
        "street": "Spuistraat",
        "numer": 210,
        "postalcode": "1012 VT",
        "city": "Amsterdam"
    }
}
```

## Suggestion to solve this

I have been experimenting with [libpostal](https://github.com/openvenues/libpostal) which is a C library which ports to Go as shown [here](https://github.com/openvenues/gopostal).

**Comment by etiennedi** [comment]
Makes sense to me. We should probably allow the string for automatic parsing **or** an object in case the user already has their data structured.

**Comment by bobvanluijt** [comment]
> We should probably allow the string for automatic parsing **or** an object in case the user already has their data structured.

That makes sense.

**Comment by etiennedi** [comment]
Do we have a way to automatically translate addresses to geoCoordinates? Possibly via third-party API integration? I.e. the user could run a command to enrich all fields of type address and in the background this would either ask the Google Maps or OpenStreetMaps API if configured.

What other benefits will the user have from using an address object over primitives?

I could imagine:

* Auto-adding geoCoordinates
* Auto adding missing properties (for example for the one above, the country is missing, but we could probably find out quite easily that it's Netherlands)

**Comment by bobvanluijt** [comment]
> Possibly via third-party API integration?

This is something I would like to do via Weaviate. Moving this specific conversation to Slack.

**Comment by etiennedi** [comment]
Closing for now, as we have identified the phoneNumber to be the most important feature which was implemented in #1088
